buildSrc/build.gradle

```diff
@@ -156,4 +156,11 @@ if (project != rootProject) {
     testClass = 'org.elasticsearch.test.NamingConventionsCheckBadClasses$UnitTestCase'
     integTestClass = 'org.elasticsearch.test.NamingConventionsCheckBadClasses$IntegTestCase'
   }
+
+  task(namingConventionsMain, type: org.elasticsearch.gradle.precommit.NamingConventionsTask) {
```

Comment:
nit: can you use the style of task creation used throughout the rest of the build.gradle files, ie:
```
task namingConventionsMain(type: org.elasticsearch.gradle.precommit.NamingConventionsTask) {
```